src/main/scala/it/pagopa/interop/authorizationprocess/server/impl/Main.scala

```diff
@@ -1,58 +1,107 @@
-
-      val serverBinding = for {
-        jwtReader <- getJwtValidator()
-        controller = new Controller(
-          clientApi(jwtReader, blockingEc),
-          healthApi,
-          operatorApi(jwtReader, blockingEc),
-          validationExceptionToRoute.some
-        )(actorSystem.classicSystem)
-        binding <- Http()(actorSystem.classicSystem)
-          .newServerAt("0.0.0.0", ApplicationConfiguration.serverPort)
-          .bind(corsHandler(controller.routes))
-      } yield binding
-
-      serverBinding.onComplete {
-        case Success(b) =>
-          logger.info(s"Started server at ${b.localAddress.getHostString()}:${b.localAddress.getPort()}")
-        case Failure(e) =>
-          actorSystem.terminate()
-          logger.error("Startup error: ", e)
-      }
-
-      Behaviors.empty[Nothing]
-    },
-    BuildInfo.name
+  implicit val context: List[(String, String)] = (CORRELATION_ID_HEADER -> UUID.randomUUID().toString()) :: Nil
+
+  implicit val actorSystem: ActorSystem[Nothing]  =
+    ActorSystem[Nothing](Behaviors.empty, "interop-be-authorization-process-alignment")
+  implicit val executionContext: ExecutionContext = actorSystem.executionContext
+
+  implicit val es: ExecutorService = Executors.newFixedThreadPool(1.max(Runtime.getRuntime.availableProcessors() - 1))
+  implicit val blockingEc          = ExecutionContext.fromExecutor(es)
+  implicit val authorizationManagementService: AuthorizationManagementService = authorizationManagementService(
+    blockingEc
   )
+  implicit val partyManagementService: PartyManagementService                 = partyManagementService()
+
+  logger.info("Starting update")
+  logger.info(s"Retrieving clients")
+  Await.result(
+    execution()
+      .andThen { case Failure(ex) => logger.error("Houston we have a problem", ex) }
+      .andThen { _ =>
+        es.shutdown()
+      },
+    Duration.Inf
+  ): Unit
+
+  logger.info("Completed update")
+
+  def execution(): Future[Unit] = for {
+    clients <- getClients()
+    _ = logger.info(s"Start update clients ${clients.size}")
+    _ <- clients.traverse(updateClient)
+    _ = logger.info(s"End update clients")
+    _ = logger.info(s"Retrieving keys")
+    keys <- clients.traverse(getClientKeys).map(_.flatten)
+    _ = logger.info(s"Start update keys ${keys.size}")
+    _ <- keys
+      .flatMap(client =>
+        client.keys.collect { case PersistentKey(Some(relationshipId), _, kid, _, _, _, _, _) =>
+          Parameter(client.id, kid, relationshipId)
+        }
+      )
+      .traverse(keys => updateKeys(keys))
+    _ = logger.info(s"End update keys")
+  } yield ()
+
+  def updateClient(client: PersistentClient): Future[Unit] = {
+    logger.info(s"Update client ${client.id}")
+    for {
+
+      relationship <- client.relationships.toList.traverse(partyManagementService.getRelationshipById)
+      _            <- relationship.traverse(rel => authorizationManagementService.addUser(client.id, rel.from))
```

Comment:
I suggest to add the user only if the client does not already contain it.
The idempotence is useful in case we need to run this script more than once (for example, if during the execution it fails for any reason)